src/vmm/src/resources.rs

```diff
@@ -1359,7 +1283,20 @@ mod tests {
             vm_resources.update_vm_config(&aux_vm_config),
             Err(VmConfigError::InvalidVcpuCount)
         );
+        aux_vm_config.smt = Some(true);
+        #[cfg(target_arch = "aarch64")]
+        assert_eq!(
+            vm_resources.update_vm_config(&aux_vm_config),
+            Err(VmConfigError::SmtNotSupported)
+        );
```

Comment:
Ah, yeah, this part could use some more comments. Its specifically about returning "invalid vcpu count" in the case where SMT is also enabled (because in that case, we have the additional condition that vcpu count needs to be even)

Reply:
Yeah, that is true on x86. My point was having another section for SMT test on AArch64.

Reply:
Done, please have another look :)